components/playlist/browser/resources/media_detector/index.js

```diff
@@ -115,25 +113,18 @@
   }
 
   function getMediaDurationInSeconds (node) {
-    let duration = node.duration
-    const isNan = (value) => { return typeof value === 'number' && Number.isNaN(value) }
-    const isInfinite = (value) => { return typeof value === 'number' && (value === Infinity || value === -Infinity) }
     const clampDuration = (value) => {
-      if (isNan(value)) {
-        return 0.0
-      }
-
-      if (isInfinite(value)) {
-        return Number.MAX_VALUE
-      }
-      return value
+      if (Number.isFinite(value)) return value
+      if (value === Number.POSITIVE_INFINITY) return Number.MAX_VALUE
+      if (value === Number.NEGATIVE_INFINITY) return Number.MIN_VALUE
+      return 0.0
     }
```

Comment:
@fallaciousreasoning Sorry but could you take a look at this part again? I've updated to use `Number`